starfish/test/full_pipelines/cli/_base_cli_test.py

```diff
@@ -0,0 +1,34 @@
+import os
+import shutil
+import unittest
+from typing import List, Tuple
+
+from starfish.intensity_table import IntensityTable
+from starfish.util import exec
+
+
+class CLITest(unittest.TestCase):
+    """This is a base class for testing CLI methods. Each stage should correspond
+    to a different pipeline step. Running the test will go through each stage and
+    run the command line method with given arguments. The last stage should produce
+    a file called results containing an IntensityTable. Each cli test should define
+    it's own verify_results method.
+    """
+    # Since this is a base class we don't want to actually run it as a test.
+    # #For classes that inherit CLITest this will need to be set to True.
+    __test__ = False
+
+    SUBDIRS = Tuple[str]
+    STAGES = Tuple[List]
+    spots_file = "spots.nc"
+
+    def verify_results(self, intensities):
+        raise NotImplementedError()
+
+    def test_run_pipline(self):
+        tempdir = exec.stages(self.STAGES, self.SUBDIRS, keep_data=True)
+        intensities = IntensityTable.load(os.path.join(tempdir, "results", self.spots_file))
+        self.verify_results(intensities)
```

Comment:
Nice. re:

 * https://github.com/spacetx/starfish/pull/516#issuecomment-419557387
 * specifically https://github.com/joshmoore/starfish/commit/d2d028499a31c3049bb9f08192e2dbeab0bf63c8#diff-3ab8577a09cb5f1f0d1527d305d08bd4

if you can see a way forward how to pass your `verify_results` _into_ `exec.stages` (perhaps as a post-processing stage that takes _only_ `tempdir` as input) then the `shutil.rmtree` block could be removed. Note: on review, @ttung asked me to change this from rmtree to a use of `TemporaryDirectory`. That's currently in https://github.com/spacetx/starfish/pull/538/commits/6a590d6ab7bde830514bb686466ee55453cf892f